test/test_strategies.rb

```diff
@@ -26,6 +26,7 @@ def file_blob(name)
 
   def all_xml_fixtures(file="*")
     fixs = Dir.glob("#{samples_path}/XML/#{file}") -
+             ["#{samples_path}/XML/psd-data.xmp"] -
```

Comment:
I don't think this is needed

Reply:
@lildude I'm not sure what this test is doing, but it fails if I comment/remove this line:

~~~console
λ linguist (xmp): nvi test/test_strategies.rb
λ linguist (xmp): brake samples test
cp lib/linguist/samples.json tmp/x86_64-darwin20/stage/lib/linguist/samples.json
install -c tmp/x86_64-darwin20/linguist/3.0.0/linguist.bundle lib/linguist/linguist.bundle
cp tmp/x86_64-darwin20/linguist/3.0.0/linguist.bundle tmp/x86_64-darwin20/stage/lib/linguist/linguist.bundle
Run options: --seed 51653

# Running:

................F...........................................................................................................................................................................................................F.................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................F..........................

Finished in 17.726060s, 81.7441 runs/s, 1678.6020 assertions/s.

  1) Failure:
TestFileBlob#test_mime_type [/Users/Alhadis/Forks/linguist/test/test_file_blob.rb:31]:
Expected: "application/pdf"
  Actual: "application/postscript"

  2) Failure:
TestStrategies#test_xml [/Users/Alhadis/Forks/linguist/test/test_strategies.rb:188]:
/Users/Alhadis/Forks/linguist/samples/XML/psd-data.xmp not detected as XML.
Expected: #<Linguist::Language name=XML>
  Actual: nil

  3) Failure:
TestBlob#test_mime_type [/Users/Alhadis/Forks/linguist/test/test_blob.rb:17]:
Expected: "application/pdf"
  Actual: "application/postscript"

1449 runs, 29755 assertions, 3 failures, 0 errors, 0 skips
rake aborted!
Command failed with status (1)
/Users/Alhadis/Forks/linguist/vendor/gems/ruby/3.0.0/gems/rake-13.0.3/exe/rake:27:in `<top (required)>'
/usr/local/opt/ruby/bin/bundle:23:in `load'
/usr/local/opt/ruby/bin/bundle:23:in `<main>'
Tasks: TOP => test
(See full trace by running task with --trace)
~~~

Reply:
I assume it's testing the XML header strategy, which fails because the sample doesn't have an XML declaration (and XMP files don't have one either).

Reply:
> I assume it's testing the XML header strategy, which fails because the sample doesn't have an XML declaration (and XMP files don't have one either).

Ah yes, that's exactly it. I see that now I look into it.

Reply:
I really, really wish we could redo our entire test suite...

I'm a stickler for high-quality test coverage, so maybe the vague-sounding test names are bugging me more than your average person. 🤷